### PR TITLE
M6 IMPL - PR #3 Flight reassignment, overload pressure, console controller, config

### PR DESCRIPTION
**Pull request by Sidarthapati**
# Pull Request
## Summary

PR #3 of M7 (hub): the hub survives the messy day — it executes M9's flight reassignments, back-pressures on overload, and completes the operator console. Reschedules become M9-decided / M7-executed (M7-D-006), and the bag now leaves the hub a configurable lead time before flight departure.

---
## What Changed

- Flight reassignment executor (§9): BagReassignmentService re-poIGHT_REASSIGNED event (keyed on flight number; coversoptimisation/cancellation/capacity/prepone-overflow via reason) — append-only item move, superseding manifest, emptied source bag → new CANCELLED status (frees its stand), emits BAG_RESCHEDULED. No M7-side SLA gate or next-fl
- FlightEventConsumer on oneday.flight.events (autoStartup=false, M9 unbuilt): FLIGHT_REASSIGNED → executor; advisory FLIGHT_TIME_CHANGED → seal-window shift
only. POST /hub/{hubId}/bags/reassign-flight is the ops/test door
- Overload back-pressure (§11): HubLoadService rolls hub_load_snapshot and emits HUB_OVERLOAD_ALERT (= M4's advisory throttle in v1) past standHighWaterPct; GET
/hub/{hubId}/load.
- Operator console: GET /hub/{hubId}/bags?date= (live directory) and POST /hub/{hubId}/parcels/{id}/resolve (ParcelLocatorService).
- Config: bag cutoff = flight departure − hubDepartureLeadMinutesthe underweight threshold and nextFlightAfter —FlightAssignmentPort is now read-only.
- Common contract: FlightEventType.FLIGHT_REASSIGNED/FLIGHT_TIME_, FlightReassignedEvent/FlightTimeChangedEvent (M9's futurecontract), BagRescheduledEvent, HubOverloadAlertEvent; M4's FlightEventsConsumer switch updated to ignore the two new M7-facing types.

---
## Why This Change?

A flight move is an economic decision (fare, booked volume, remaiat only M9 has — M7 sees only weight, so an M7-side reschedulewould fight M9's optimisation. PR3 makes reschedule purely M9-decided and M7-executed, adds the hub's overload safety net (escalate, never silently drop an
SLA), and finishes the console. The lead-time config reflects theave the hub well before departure (airport transfer + GHA handover + loading).

---
## Testing

- [x] Tested locally
- [x] Edge cases considered
- [x] No breaking changes introduced

## Test Evidence

Hub suite (34, up from 26) + routing (56) green; full app assembl
[INFO] Tests run: 3,  Failures: 0, Errors: 0 -- BagReassignmentServiceImplTest
[INFO] Tests run: 2,  Failures: 0, Errors: 0 -- HubLoadServiceImp
[INFO] Tests run: 1,  Failures: 0, Errors: 0 -- StubFlightAssignmentPortTest
[INFO] Tests run: 7,  Failures: 0, Errors: 0 -- HubEventPayloadJa
[INFO] Tests run: 9,  Failures: 0, Errors: 0 -- FlightBagServiceImplTest
[INFO] Tests run: 5,  Failures: 0, Errors: 0 -- HubReceivingServi
[INFO] Tests run: 4,  Failures: 0, Errors: 0 -- DeliveryBagServiceImplTest
[INFO] Tests run: 3,  Failures: 0, Errors: 0 -- SortServiceImplTe
[INFO] Tests run: 34, Failures: 0, Errors: 0, Skipped: 0
[INFO] BUILD SUCCESS
--- routing ---
[INFO] Tests run: 56, Failures: 0, Errors: 0, Skipped: 0
[INFO] BUILD SUCCESS
--- assemble ---
mvn clean install -pl common,orders,grid,routing,hub,app -am -DskipTests → BUILD SUCCESS
Coverage: whole-flight cancellation → source CANCELLED; subset opN; nothing-resolved → 422; overload over/under high-water; cutoff= departure − 5h; FLIGHT_REASSIGNED/BAG_RESCHEDULED Jackson round-trips.

---
## Impact

Areas Affected

- [x] Backend
- [ ] Frontend
- [ ] Routing
- [ ] Infra
- [ ] Database
- [ ] NA

## Modules Affected

- [ ] M1 — Auth
- [ ] M2 — Pricing
- [ ] M3 — Grid
- [x] M4 — Orders
- [ ] M5 — Dispatch
- [ ] M6 — Routing
- [x] M7 — Hub
- [ ] M8 — Barcode
- [x] M9 — Airline
- [ ] M10 — SLA
- [ ] M11 — Exceptions
- [ ] NA

---
## Risks / Concerns

- Low. FlightEventConsumer is dormant (autoStartup=false) until M9 produces, so the reschedule path has no runtime effect yet; it's exercised via the REST
endpoint + unit tests. The new FlightBagStatus.CANCELLED value wrth no DB CHECK constraint, so no migration is needed. Append-onlyintegrity is preserved (items flip to REMOVED, manifests are superseded, never mutated). Unrelated pre-existing auth UserManagementE2eTest failures (schema
mismatch: missing column [phone] in onboarding_requests) surface  not introduced by this PR.

---
## Checklist

- [x] Code follows project conventions (package layout, no cross-module internal imports)
- [x] Self-review completed
- [x] Append-only audit trail preserved (no mutations to scan/manifest/role-change records)
- [x] No sensitive data or credentials exposed
- [x] Documentation updated if behaviour changed
- [x] Ready for review

### CODE REVIEW

File: hub/src/main/java/com/oneday/hub/service/impl/BagReassignmentServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+        FlightBag sourceTemplateBag = requireBag(template.getBagId());
+        FlightBag target = flightBagService.openBag(new FlightBagService.OpenBagCommand(
+                sourceTemplateBag.getCityId(), sourceTemplateBag.getHubId(),
+                cmd.toFlightNo(), cmd.toFlightDate(), sourceTemplateBag.getOriginHub(),
+                cmd.destHub(), cmd.newCutoff()));
+
+        Set<UUID> affectedSourceBagIds = new LinkedHashSet<>();
+        UUID supersedesManifestId = null;
+        int moved = 0;
+        for (FlightBagItem item : sourceItems) {
```
Comment (Sidarthapati) [inline hub/src/main/java/com/oneday/hub/service/impl/BagReassignmentServiceImpl.java:76]: Why are we relying on the parcelIds from the api response, shouldnt we have a repo query for it, for all parcels in that flight bag, we already know these.

File: hub/src/main/java/com/oneday/hub/service/impl/BagReassignmentServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+            return items;
+        }
+        if (cmd.fromFlightNo() == null) {
+            return List.of();
+        }
+        return flightBagRepository
+                .findFirstByFlightNoAndDestHubAndStatusInOrderByCreatedAtDesc(cmd.fromFlightNo(), cmd.destHub(),
+                        List.of(FlightBagStatus.OPEN, FlightBagStatus.SEALED))
+                .map(bag -> flightBagItemRepository.findByBagIdAndStatus(bag.getId(), FlightBagItemStatus.IN_BAG))
+                .orElseGet(List::of);
```
Comment (Sidarthapati) [inline hub/src/main/java/com/oneday/hub/service/impl/BagReassignmentServiceImpl.java:125]: Got it so we are taking care if we do no get parcelIds in response here only.
